snuba/datasets/message_filters.py

```diff
@@ -46,6 +46,28 @@ def should_drop(self, message: Message[KafkaPayload]) -> bool:
         return False
 
 
+@dataclass
+class KafkaHeaderSelectFilter(StreamMessageFilter[KafkaPayload]):
+    """
+    A filter over messages coming from a stream which matches whether the given message
+    has the provided header key and if it matches the provided header value. If there is
+    a match, the message gets processed. (This is the inverse of KafkaHeaderFilter)
+    """
+
+    header_key: str
+    header_value: str
+
+    def should_drop(self, message: Message[KafkaPayload]) -> bool:
+        for key, value in message.payload.headers:
+            if key != self.header_key:
+                continue
+
+            str_value = value.decode("utf-8")
+            return False if str_value == self.header_value else True
```

Comment:
are you assuming that the same header key is likely to appear twice? because in this clause we've already detected that the key we're inspecting is the key of concern